sklearn/utils/_repr_html/params.py

```diff
@@ -2,11 +2,64 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 import html
+import re
 import reprlib
 from collections import UserDict
+from urllib.parse import quote
 
 from sklearn.utils._repr_html.base import ReprHTMLMixin
 
+CLASS_DOC_URL_PREFIX = "https://scikit-learn.org/{doc_version}/modules/generated/"
+
+
+def link_to_param_doc(estimator_type, param_name):
+    """URL to the relevant section of the docstring using a Text Fragment
+
+    https://developer.mozilla.org/en-US/docs/Web/URI/Reference/Fragment/Text_fragments
+    """
+
+    import sklearn
+
+    if hasattr(estimator_type, "__module__"):
+        module_name = estimator_type.__module__
+    else:
+        module_name = None
+
+    if module_name is None or not module_name.startswith("sklearn."):
+        # Not a scikit-learn estimator. Do not link to the scikit-learn
+        # documentation.
+        return None
+
+    if ".dev" in sklearn.__version__:
+        doc_version = "dev"
+    else:
+        doc_version = ".".join(sklearn.__version__.split(".")[:2])
+
+    class_doc_base_url = CLASS_DOC_URL_PREFIX.format(doc_version=doc_version)
+
+    # Strip private submodule component if any:
+    if "._" in module_name:
+        module_name = module_name.split("._")[0]
+
+    class_name = estimator_type.__qualname__
+
+    docstring = estimator_type.__doc__
+
+    m = re.search(f"{param_name} : (.+)\\n", docstring)
+    if m is None:
+        # No match found in the docstring, return None to indicate that we
+        # cannot link.
+        return None
+
+    # Extract the first word of the type information as disambiguation suffix
+    # to build the fragment
+    param_type = m.group(1)
```

Comment:
the way the search is constructed, it's not the first word, but the whole line up to the line break.

Reply:
Oh yes.. I changed the regex to make it work, and forgot to change the comment there.